File: 311__Sparse_Matrix_Multiplication.py
class Solution(object):
    def multiply(self, A, B):
        """
        :type A: List[List[int]]
        :type B: List[List[int]]
        :rtype: List[List[int]]
        """
        result = [[0] * len(B[0]) for _ in range(len(A))]

        for i in range(len(A)):
            for j in range(len(A[0])):
                if A[i][j] != 0:
                    for k in range(len(B[j])):
                        result[i][k] += A[i][j] * B[j][k]
        return result
        

File: Java/311__Sparse_Matrix_Multiplication.java
class Solution {
    public int[][] multiply(int[][] A, int[][] B) {
        if (A == null || A.length == 0 || A[0].length == 0 || B == null || B.length == 0 || B[0].length == 0)
            return null;
        int m = A.length, n = A[0].length, p = B[0].length;
        int[][] result = new int[m][p];
        for (int j = 0; j < p; j++) {
            for (int i = 0; i < m; i++) {
                for (int temp = 0; temp < n; temp++) {
                    result[i][j] += A[i][temp] * B[temp][j];
                }
            }
        }
        return result;
    }
}
